ejb/ejb2-connector/ejb2-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/custom/connector/EJBUtil.java

```diff
@@ -0,0 +1,193 @@
+import java.util.Properties;
+import java.util.Set;
+
+public class EJBUtil {
+    private static final Log log = LogFactory.getLog(EJBUtil.class);
+
+    /**
+     * @param instance instance of an ejb object
+     * @param method   method that we want to call
+     * @param args     arguments object for method
+     * @return invoke arguments into method and return the value which actual method returns
+     */
+    public static Object invokeInstanceMethod(Object instance, Method method, Object[] args) {
+        Class[] paramTypes = method.getParameterTypes();
+        if (paramTypes.length != args.length) {
+            handleException("Provided argument count does not match method the "
+                    + "parameter count of method '" + method.getName() + "'. Argument count = "
+                    + args.length + ", method parameter count = " + paramTypes.length);
+        }
+        Object[] processedArgs = new Object[paramTypes.length];
+        for (int i = 0; i < paramTypes.length; ++i) {
+            if (args[i] == null || paramTypes[i].isAssignableFrom(args[i].getClass())) {
+                processedArgs[i] = args[i];
+            } else if (SimpleTypeMapper.isSimpleType(paramTypes[i])) {
+                try {
+                    processedArgs[i] = SimpleTypeMapper.getSimpleTypeObject(paramTypes[i],
+                            AXIOMUtil.stringToOM("<a>" + args[i] + "</a>"));
+                } catch (XMLStreamException e) {
+                    handleException("XMLStreamException ", e);
+                }
+            } else {
+                handleException("Incompatible argument found in " + i + "th argument "
+                        + "for '" + method.getName() + "' method.");
+            }
+        }
+        try {
+            return method.invoke(instance, processedArgs);
+        } catch (IllegalAccessException e) {
+            handleException("Error while invoking '" + method.getName() + "' method "
+                    + "via reflection.", e);
+        } catch (InvocationTargetException e) {
+            handleException("Error while invoking '" + method.getName() + "' method "
+                    + "via reflection.", e);
+        }
+        return null;
+    }
+
+    /**
+     * @param aClass     class of our target ejb remote
+     * @param methodName name of the method
+     * @param argCount   number of arguments
+     * @return extract the value's from properties and make its as hashable
+     */
+    public static Method resolveMethod(Class aClass, String methodName, int argCount) {
+        Method resolvedMethod = null;
+        for (Method method : aClass.getDeclaredMethods()) {
+            if (method.getName().equals(methodName) &&
+                    method.getParameterTypes().length == argCount) {
+                if (resolvedMethod == null) {
+                    resolvedMethod = method;
+                } else {
+                    handleException("More than one '" + methodName + "' methods " +
```

Comment:
yes,
since we are checking the number of parameter for find the method in remote object some times some methods with same name and same number of parameters may available, so to avoid this situation we need this catch.